api/integrations/lead_tracking/hubspot/lead_tracker.py

```diff
@@ -44,88 +45,134 @@ def should_track(user: FFAdminUser) -> bool:
 
         return True
 
-    def create_lead(self, user: FFAdminUser, organisation: Organisation = None) -> None:  # type: ignore[assignment]
-        contact_data = self.client.get_contact(user)
-
-        if contact_data:
-            # The user is already present in the system as a lead
-            # for an existing organisation, so return early.
-            return
+    def create_user_hubspot_contact(self, user: FFAdminUser) -> str | None:
+        tracker = HubspotTracker.objects.filter(user=user).first()
+        tracker_cookie = tracker.hubspot_cookie if tracker else None
+        self.client.create_lead_form(user=user, hubspot_cookie=tracker_cookie)
+        contact = self.client.get_contact(user)
+        hubspot_contact_id: str | None = contact.get("id")
+
+        # Hubspot creates contact asynchronously
+        # If not available on the spot, following steps will sync database with Hubspot
+        if hubspot_contact_id is not None:
+            HubspotLead.objects.update_or_create(
+                user=user, defaults={"hubspot_id": hubspot_contact_id}
+            )
```

Comment:
Discussed with Matt on this case.
Basically, it's a matter of synchronization. Contact is created, but our `get_contact` request *might* be sent before processed by hubspot. When associating a user with an organisation, there is a fallback to still check his existence in hubspot so that will resynchronize us with hubspot anyways. Only user creating an account and no organization will be "zombies"
As we use the hubspot_id only for association and marketing does not really need it, it's an accepted trade-off

Alternatives were:
- Top notch solution would be webhook but overkill
- Retry mechanism